x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/filestructurefinder/GrokPatternCreator.java

```diff
@@ -486,9 +524,17 @@ public String processCaptures(Map<String, Integer> fieldNameCountStore, Collecti
                 Map<String, String> fullMappingType = Collections.singletonMap(FileStructureUtils.MAPPING_TYPE_SETTING, mappingType);
                 if ("date".equals(mappingType)) {
                     assert values.isEmpty() == false;
-                    TimestampMatch timestampMatch = TimestampFormatFinder.findFirstFullMatch(values.iterator().next(), timeoutChecker);
-                    if (timestampMatch != null) {
-                        fullMappingType = timestampMatch.getEsDateMappingTypeWithFormat();
+                    TimestampFormatFinder timestampFormatFinder = new TimestampFormatFinder(explanation, true, true, true, timeoutChecker);
```

Comment:
This for loop all the way to calling `timestampFormatFinder.getEsDateMappingTypeWithFormat()` is repeated in `FileStructureUtils` around line `277`. I wonder if we could refactor that in a method.